File: calorie-log-web/src/api/food.ts
import { apiGet, apiPost } from './client';
import type { Food, PageResult } from '../types';

export const searchFood = (keyword: string, page = 1, size = 20) =>
  apiGet<PageResult<Food>>('/foods/search', { keyword, page, size });

export const getFood = (id: number) => apiGet<Food>(`/foods/${id}`);

export interface CreateFoodBody {
  name: string;
  alias?: string;
  category?: string;
  calories: number;
  protein?: number;
  carbohydrate?: number;
  fat?: number;
  dietaryFiber?: number;
  addedSugar?: number;
  isHardToWeigh?: boolean;
  grossNetRatio?: number;
  barcode?: string;
}

export const createCustomFood = (body: CreateFoodBody) =>
  apiPost<Food>('/foods/custom', body);

File: calorie-log-server/src/main/java/com/calorielog/module/food/entity/Food.java
package com.calorielog.module.food.entity;

import com.baomidou.mybatisplus.annotation.FieldFill;
import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableLogic;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Data
@TableName("t_food")
public class Food {
    @TableId(type = IdType.AUTO)
    private Long id;
    private String name;
    private String alias;
    private String barcode;
    private String category;
    private String unit;
    private BigDecimal calories;
    private BigDecimal protein;
    private BigDecimal carbohydrate;
    private BigDecimal fat;
    private BigDecimal dietaryFiber;
    private BigDecimal addedSugar;
    private BigDecimal vitaminA;
    private BigDecimal vitaminB1;
    private BigDecimal vitaminB2;
    private BigDecimal vitaminC;
    private BigDecimal vitaminE;
    private BigDecimal sodium;
    private BigDecimal potassium;
    private BigDecimal calcium;
    private BigDecimal iron;
    private BigDecimal zinc;
    private Boolean isHardToWeigh;
    private BigDecimal grossNetRatio;
    private String dataSource;
    private Long createdBy;

    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createdAt;
    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updatedAt;

    @TableLogic(value = "null", delval = "CURRENT_TIMESTAMP")
    @TableField(select = false)
    private LocalDateTime deletedAt;
}
